app/models/katello/glue/candlepin/candlepin_object.rb

```diff
@@ -5,8 +5,7 @@ module Glue::Candlepin::CandlepinObject
     module ClassMethods
       def get_for_organization(organization)
         # returns objects from AR database rather than candlepin data
-        pool_ids = self.get_for_owner(organization.label).collect { |x| x['id'] }
-        self.where(:cp_id => pool_ids)
+        self.where(:organization => organization)
```

Comment:
It looks like this method is only used in one place, and I don't think it makes sense to have a separate method anymore since it only has a single `where` clause.

Let's remove this method and update subscriptions_controller.rb:

```
+++ b/app/controllers/katello/api/v2/subscriptions_controller.rb
@@ -65,7 +65,7 @@ module Katello
       return available_for_activation_key if params[:available_for] == "activation_key"
       collection = Pool.readable
       collection = collection.where(:unmapped_guest => false)
-      collection = collection.get_for_organization(Organization.find(params[:organization_id])) if params[:organization_id]
+      collection = collection.where(organization: Organization.find(params[:organization_id])) if params[:organization_id]
       collection = collection.for_activation_key(@activation_key) if params[:activation_key_id]
       collection
     end
```

I think the tests will need to be updated. You can verify locally with:
```
ktest ./test/controllers/api/v2/subscriptions_controller_test.rb
```

Reply:
Thanks, Jonathon. This is exactly what I came out first. However, I was worried about breaking the tests as they call get_for_organization. I'll take a look further for that.